docs/adr/0002-schema-first-api-design.md

```diff
@@ -0,0 +1,59 @@
+
+## Afspraken
+
+- Een externe API contract (wordt geconsumeerd door applicaties van zowel binnen als buiten de organisatie van de provider) wordt beheerd in een voor dat API opgezette GitHub repo. Een voorbeeld van zo'n repo is de [Haal Centraal BRP bevragen][ApiContractRepo] repo. De benodigde OAS bestanden van/voor het API contract en het gepubliceerde (in genereervariant map) OAS bestand van het API contract staan in de **specificatie** map van de repo.
+De repo waar de provider implementatie van het externe API contract wordt beheerd, bevat in zijn **specificatie** map van de **main** branch een kopie van dé versie van het OAS bestand van het externe API contract dat door de provider is geïmplementeerd. Een voorbeeld van zo'n repo is de [personen-informatie-service][InformatieServiceRepo] repo.
+- Een interne API contract (kan alleen door applicaties binnen de organisatie van de provider worden geconsumeerd) wordt beheerd in de GitHub repo waar de provider implementatie van het API contract wordt beheerd. Voorbeelden van zo'n repo zijn de [personen-data-service][DataServiceRepo] repo en de [brp-api-gezag][GezagService] repo. De benodigde OAS bestanden van/voor een interne API contract staan ook in de **specificatie** map van de repo.
+- Wijzigingen aan een externe API contract worden gedaan in een branch in de repo van het externe API contract. Wanneer de wijzigingen aan het API contract zijn goedgekeurd, dan wordt de nieuwe gepubliceerde versie van het OAS bestand handmatig/automatisch gekopieerd naar de repo van de provider van het API contract. Om de relatie tussen de wijzigingen in de verschillende GitHub repos te kunnen leggen wordt over al de repos dezelfde branch naam gebruikt
+
+Onderstaand sequence diagram illustreert de hierboven beschreven afspraken
+```mermaid
+sequenceDiagram
+    actor consumer
+    consumer-->>ApiContractRepo: request API feature 'A'
+    ApiContractRepo-->>ApiContractRepo: create branch 'feature A'
+    ApiContractRepo-->>ApiContractRepo: design API contract
+    ApiContractRepo-->>InformatieServiceRepo: sync approved API contract
+    InformatieServiceRepo-->>InformatieServiceRepo: create branch 'feature A'
+    InformatieServiceRepo-->>DataServiceRepo: request API wijziging voor feature 'A'
+    DataServiceRepo-->>DataServiceRepo: create branch 'feature A'
+    DataServiceRepo-->>DataServiceRepo: design API contract
+    InformatieServiceRepo-->>InformatieServiceRepo: code generatie
+    DataServiceRepo-->>DataServiceRepo: code generatie
+    InformatieServiceRepo-->>InformatieServiceRepo: implementeer logic
+    DataServiceRepo-->>DataServiceRepo: implementeer logic
+    ApiContractRepo-->>ApiContractRepo: merge branch 'feature A' naar main
+    InformatieServiceRepo-->>InformatieServiceRepo: merge branch 'feature A' naar main
+    DataServiceRepo-->>DataServiceRepo: merge branch 'feature A' naar main
+```
```

Comment:
Begrijp ik het goed dat we starten met het product voor de klant en eindigen met de implementatie van de logic in de informatie- en dataservice. Het proces gaat van klant(vraag) naar implementatie(detail). En implementatie kan (daardoor) gelijktijdig gebeuren.

Reply:
klopt. Ik heb deze opmerking verwerkt in het diagram